src/java.base/share/native/libjava/check_version.c

```diff
@@ -30,16 +30,5 @@
 JNIEXPORT jint JNICALL
 DEF_JNI_OnLoad(JavaVM *vm, void *reserved)
 {
-    jint vm_version = JVM_GetInterfaceVersion();
-    if (vm_version != JVM_INTERFACE_VERSION) {
-        JNIEnv *env;
-        char buf[128];
-        sprintf(buf, "JVM interface version mismatch: expecting %d, got %d.",
-                JVM_INTERFACE_VERSION, (int)vm_version);
-        (*vm)->GetEnv(vm, (void **)&env, JNI_VERSION_1_2);
-        if (env) {
-            (*env)->FatalError(env, buf);
-        }
-    }
     return JNI_VERSION_1_2;
```

Comment:
This leaves an entire file with one trivial function implementation. Can we remove the file and implement  `DEF_JNI_OnLoad()` in `jni_util.h` ?

Reply:
I am not sure if jni_utils.c is the right file (it defines the `JNU_XXX` functions that are used by other shared libraries).

There are other .c files that have trivial `DEF_JNI_OnLoad` functions (e.g., java.base/share/native/libnio/nio_util.c).

@AlanBateman do you have any suggestions?

Reply:
I'm fine with the way it is, just thought we might want to consider cleaning up a bit more, since it's a cleanup task itself.

Reply:
Thanks Gerard. The main purpose of this PR is to clean up the JVM side. I'll leave the refactoring of check_version.c to the core-lib team.